### ISSUE TITLE
Add tool dependencies

### ISSUE THREAD

**Issue by ian-howell** [kubernetes-sigs/kustomize#1226]
With the advent of go modules, we now have the ability to add dependencies on tools. Right now, a new contributor to Kustomize won't be able to successfully run all of the tests due to several missing tools (`golangci-lint`, `mdrip`, `kubeval`, etc...). I propose following the guidelines laid out [here](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module) to streamline the process.

This would involve creating a new file at `kustomize/tools/tools.go`, whose contents might look something like this:
```go
// +build tools
package tools

import (
	_ "github.com/golangci/golangci-lint/cmd/golangci-lint"
	_ "github.com/monopole/mdrip"
        _ "github.com/instrumenta/kubeval"
)
```

**Comment by arnodel** [comment]
I submitted a fix for #1228 yesterday and I struggled a bit when trying to test my patch.  In fact even some markdown page documenting the process would be helpful.  What I did was to install the tools similarly to what is specified in `.travis.yml` and then run `travis/pre-commit.sh`.  That might be obvious to some but it wasn't to me!

**Comment by ian-howell** [comment]
I'm going to go ahead and do some work on this change, but it seems like it will directly conflict with some of the stuff in [the travis pre-commit script](https://github.com/kubernetes-sigs/kustomize/blob/master/travis/pre-commit.sh).

@monopole @Liujingfang1 Do either of you guys have any opposition to this change? I think this will dramatically smooth out the process for community developers.
